### PR TITLE
Update Kubernetes libraries to v0.20.4

### PR DESCRIPTION
**Pull request by michaelmdresser**
Context needed to be added to a few API calls as a result. This upgrade
was performed by running:

`go get k8s.io/client-go@v0.20.4`

And then resolving compiler errors by adding `context.Background()` where needed.

Also updates `compat.go` to prefer the stable region label and not hardcode it. I opted to not update the instance type label at this time, but it should eventually prefer `v1.LabelInstanceTypeStable`.

Goes with https://github.com/kubecost/kubecost-cost-model/pull/345

### CODE REVIEW

**Comment by michaelmdresser** [comment]
I had to update `golang.org/x/sync` and `github.com/prometheus/client_model` to get the build in CI to work (even though it was working locally with `docker build`).

**Comment by michaelmdresser** [comment]
Tested with the open source-only and no Prometheus deploy path, which is working (`/metrics` looks good, implying the API watchers are working correctly) and the logs don't indicate anything out of order.

File: pkg/cloud/csvprovider.go [hunk 1]
```diff
@@ -230,7 +230,11 @@ func NodeValueFromMapField(m string, n *v1.Node, useRegion bool) string {
 	mf := strings.Split(m, ".")
 	toReturn := ""
 	if useRegion {
-		toReturn = n.Labels[v1.LabelZoneRegion] + ","
+		if region, ok := util.GetRegion(n.Labels); ok {
+			toReturn = region + ","
+		} else {
+			klog.Infof("[ERROR] Getting region based on labels failed")
```
Comment (nikovacevic) [inline pkg/cloud/csvprovider.go:236]: Could use `log.Errorf(...)` here

File: pkg/util/compat.go [hunk 2]
```diff
@@ -4,11 +4,13 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
+// See https://kubernetes.io/docs/reference/labels-annotations-taints/
+
 func GetRegion(labels map[string]string) (string, bool) {
-	if _, ok := labels[v1.LabelZoneRegion]; ok {
+	if _, ok := labels[v1.LabelTopologyRegion]; ok { // Label as of 1.17
+		return labels[v1.LabelTopologyRegion], true
+	} else if _, ok := labels[v1.LabelZoneRegion]; ok { // deprecated label
```
Comment (nikovacevic) [inline pkg/util/compat.go:12]: Was this, specifically, tested? If so, just done manually on one of our GCP clusters?

**Review by nikovacevic — APPROVED** [review]

File: pkg/util/compat.go (hunk 2 above)
Comment (michaelmdresser) [reply pkg/util/compat.go:12]: So, this is used here: https://github.com/kubecost/cost-model/blob/mmd-update-k8s-libraries/pkg/costmodel/costmodel.go#L977

This result is, eventually, consumed by the emitter of `kubecost_node_is_spot` (among other things) which includes region info. I am seeing that accurate on a GKE cluster (showing `us-east1` for a cluster running in `us-east1-b`).

Comment (michaelmdresser) [reply pkg/util/compat.go:12]: This matches up with pre-change behavior, as far as I can tell with a fresh cluster and the regular helm chart.

Comment (michaelmdresser) [reply pkg/util/compat.go:12]: It's also used in the providers, including GCP provider: https://github.com/kubecost/cost-model/blob/mmd-update-k8s-libraries/pkg/cloud/gcpprovider.go#L1180 and I'm not seeing related errors in the logs.

File: pkg/cloud/csvprovider.go (hunk 1 above)
Comment (michaelmdresser) [reply pkg/cloud/csvprovider.go:236]: I agree, but I was copying the error log formatting elsewhere in the function. I'm in favor of switching all of them. Thoughts?

Comment (AjayTripathy) [reply pkg/cloud/csvprovider.go:236]: klog.Infof is legacy so please do migrate them where it's quick and you're in the code anyway.

Comment (michaelmdresser) [reply pkg/cloud/csvprovider.go:236]: Do we prefer stdlib `log`?

Comment (AjayTripathy) [reply pkg/cloud/csvprovider.go:236]: Sorry for the confusion. It's our own logging utility: 	"github.com/kubecost/cost-model/pkg/log"

Comment (michaelmdresser) [reply pkg/cloud/csvprovider.go]: Okay. Removed `klog` from the dependencies on `csvprovider.go` and updated the offending info logs to be error logs without the ERROR prefix.

File: pkg/util/compat.go (hunk 2 above)
Comment (nikovacevic) [reply pkg/util/compat.go:12]: Sounds good

**Review by AjayTripathy — APPROVED** [review]

**Pull request merged**